File: objc/Project7/Project7/City.m
//
//  City.m
//  Project7
//
//  Created by Jinwoo Kim on 2/11/21.
//

#import "City.h"

@implementation City

- (City *)initWithName:(NSString *)name country:(NSString *)country coordinates:(CLLocationCoordinate2D)coordinates {
    self = [self init];
    if (self) {
        self.name = name;
        self.country = country;
        self.coordinates = coordinates;
    }
    return self;
}

- (NSString *)formattedName {
    return [NSString stringWithFormat:@"%@ (%@)", self.name, self.country];
}

- (BOOL)matchesWithText:(NSString *)text {
    return ([self.name localizedCaseInsensitiveContainsString:text] || [self.country localizedCaseInsensitiveContainsString:text]);
}

- (BOOL)isEqualToCity:(City *)comparison {
    return [self.name isEqualToString:comparison.name];
}

@end

File: swift/Project7/Project7/City.swift
//
//  City.swift
//  Project7
//
//  Created by Jinwoo Kim on 2/11/21.
//

import Foundation
import CoreLocation

struct City: Comparable {
    var formattedName: String {
        return "\(name) (\(country))"
    }
    
    var name: String
    var country: String
    var coordinates: CLLocationCoordinate2D
    
    func matches(_ text: String) -> Bool {
        return name.localizedCaseInsensitiveContains(text) || country.localizedCaseInsensitiveContains(text)
    }
    
    static func < (lhs: City, rhs: City) -> Bool {
        return lhs.name < rhs.name
    }
    
    static func == (lhs: City, rhs: City) -> Bool {
        return lhs.name == rhs.name
    }
}
